docs/2-developing/0-connectors/2-connector-specification.mdx

```diff
@@ -2,25 +2,156 @@
 title: "Connector Specification"
 ---
 
-[Specification](https://pkg.go.dev/github.com/conduitio/conduit-connector-sdk#Specification) contains general information regarding the plugin like its name and what it does.
+A connector's specification is used to present information about the connector
+and validate connector parameters. The specification can be found in the
+`connectors.yaml` file, that's found in the root of a connector's repository.
+The following is an example of
+the [file connector's](https://github.com/ConduitIO/conduit-connector-file)
+`connector.yaml`:
 
-### `spec.go`
+```yaml
+version: "1.0"
+specification:
+  name: file
+  summary: A file source and destination plugin for Conduit.
+  description: |-
+    The file source allows you to listen to a local file and
+    detect any changes happening to it. Each change will create a new record. The
+    destination allows you to write record payloads to a destination file, each new
+    record payload is appended to the file in a new line.
+  version: v0.10.0
+  author: Meroxa, Inc.
+  source:
+    parameters:
+      - name: path
+        description: Path is the file path used by the connector to read/write records.
+        type: string
+        default: ""
+        validations:
+          - type: required
+            value: ""
+    # other parameters
+```
+
+The contents of this file are embedded into a connector's binary and read by the
+connector binary itself. `connector.yaml` is also being used by other tools that
+enrich Conduit's functionality, such as the CLI, the documentation website, etc.
+
+## Contents of `connector.yaml`
+
+A part of `connector.yaml` is written by a developer (manually) and the other
+part is generated by a tool that Conduit provides, [
+`specgen`](https://github.com/ConduitIO/conduit-connector-sdk/tree/main/specgen).
+
+A developer is writing the general information about the connector, which can be
+seen in this part from the example above:
+
+```yaml
+specification:
+  name: file
+  summary: A file source and destination plugin for Conduit.
+  description: |-
+    The file source allows you to listen to a local file and
+    detect any changes happening to it. Each change will create a new record. The
+    destination allows you to write record payloads to a destination file, each new
+    record payload is appended to the file in a new line.
+  version: v0.10.0
```

Comment:
before this, we had "devel" as a way to keep the connector version consistent with the specifications version, is there an alternative for this now? or is it the developer's responsibility to keep this up to date?

Reply:
"devel" was a placeholder or no? But yes, it's the developer's responsibility to keep this up to date. `make build` will update the version (while developing) and so will a script that we provide ([link](https://github.com/ConduitIO/conduit-connector-template/pull/71/files#diff-22e9e700507ebd287d3f7b0c100f1b943699b59f1af0e8a672ff2bc5f1df0f21), when preparing a release).